### DISCUSSION TITLE
article/spring-boot/spring-boot-virtual-threads-vs-webflux

### DISCUSSION THREAD

**Discussion by giscus** [dyc87112/giscus-didispace#5] [Announcements]
# article/spring-boot/spring-boot-virtual-threads-vs-webflux

Spring Boot虚拟线程与Webflux在JWT验证和MySQL查询上的性能比较 早上看到一篇关于Spring Boot虚拟线程和Webflux性能对比的文章，觉得还不错。内容较长，我就不翻译了，抓重点给大家介绍一下这篇文章的核心内容，方便大家快速阅读。 测试场景 作者采用了一个尽可能贴近现实操作的场景： 从授权头信息中提取JWT 验证JWT并从中提取用户的Email 使用用户的Email去MySQL里执行查询 返回用户记录

https://www.didispace.com/article/spring-boot/spring-boot-virtual-threads-vs-webflux.html

**Comment by wxa56** [comment]
应该再比较一下 springboot vs springboot（vt）

**Reply by wxa56** [reply]
这个测试参考意义有限
1. 根据官方文档好像不是只配一下TomcatProtocolHandlerCustomizer就可以完全启用vt新特性的（11月底spring boot新版本发布后可以再测一下）。
2. 目前整个java世界对vt的友好度还很低，目前还无法准确评估

**Comment by zhenruyan** [comment]
这其实就变成了 同步请求mysql和异步请求mysql再这里的并发情况了吧

**Comment by rowstop** [comment]
Mysql的驱动不一样，VT下的mysql的驱动仍然是同步的

**Reply by rowstop** [reply]
应该是还没兼容
